transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollSocketChannel.java

```diff
@@ -43,48 +39,32 @@
 
     private final EpollSocketChannelConfig config;
 
-    private volatile InetSocketAddress local;
-    private volatile InetSocketAddress remote;
-    private InetSocketAddress requestedRemote;
-
     private volatile Collection<InetAddress> tcpMd5SigAddresses = Collections.emptyList();
 
-    EpollSocketChannel(Channel parent, LinuxSocket fd, InetSocketAddress remote) {
-        super(parent, fd);
-        config = new EpollSocketChannelConfig(this);
-        // Directly cache the remote and local addresses
-        // See https://github.com/netty/netty/issues/2359
-        this.remote = remote;
-        local = fd.localAddress();
-
-        if (parent instanceof EpollServerSocketChannel) {
-            tcpMd5SigAddresses = ((EpollServerSocketChannel) parent).tcpMd5SigAddresses();
-        }
-    }
-
     public EpollSocketChannel() {
         super(newSocketStream(), false);
         config = new EpollSocketChannelConfig(this);
     }
 
     public EpollSocketChannel(int fd) {
         super(fd);
-        // As we create an EpollSocketChannel from a FileDescriptor we should try to obtain the remote and local
-        // address from it. This is needed as the FileDescriptor may be bound/connected already.
-        remote = socket.remoteAddress();
```

Comment:
related to https://github.com/netty/netty/pull/6193#discussion_r124035232

Just wanted to double check if this change in behavior is intended ... if the connection hasn't completed synchronously the local address will now be null.